File: SMLanguage/Models/RunningScenario.cs
using System;
using System.Collections.Concurrent;

namespace SMLanguage.Models
{
    public class RunningScenario
    {
        public Guid Id { get; set; }
        public string Name { get; set; }

        public string Background { get; set; }

        public int Line { get; set; }

        public bool Redraw { get; set; }

        public ConcurrentDictionary<Guid, Character> Characters;

        public ChoiceSelector CurrentChoiceSelector { get; set; }

        //TODO: Tree idea, maybe we can store a running tally of choices made

    }
}

File: SM-engine/src/models/RunningScenario.ts
import { Character } from './Character';
import { ChoiceSelector } from './ChoiceSelector';

export class RunningScenario {
  public Id: string = '';
  public Name: string = '';
  public fileName: string = ""; // Allows for multiple scenarios in one file
  public Line: number = 0;
  public Background: string = '';
  public Characters: Map<string, Character> = new Map();
  public Redraw: boolean = false;
  public CurrentChoiceSelector: ChoiceSelector = {
    Id: '',
    Question: '',
    Choices: {},
  };
  public ChoiceSelectors: ChoiceSelector[] = [];
}
